test/common/upstream/health_check_fuzz.cc

```diff
@@ -454,16 +454,63 @@ void GrpcHealthCheckFuzz::raiseEvent(const Network::ConnectionEvent& event_type,
 
 void GrpcHealthCheckFuzz::raiseGoAway(bool no_error) {
   if (no_error) {
-    test_sessions_[0]->codec_client_->raiseGoAway(Http::GoAwayErrorCode::NoError);
+    test_session_->codec_client_->raiseGoAway(Http::GoAwayErrorCode::NoError);
     // Will cause other events to blow away client, because this is a "graceful" go away
     received_no_error_goaway_ = true;
   } else {
     // go away events without no error flag explicitly blow away client
-    test_sessions_[0]->codec_client_->raiseGoAway(Http::GoAwayErrorCode::Other);
+    test_session_->codec_client_->raiseGoAway(Http::GoAwayErrorCode::Other);
     triggerIntervalTimer(true);
   }
 }
 
+void GrpcHealthCheckFuzz::expectSessionCreate() {
+  test_session_->timeout_timer_ = new NiceMock<Event::MockTimer>(&dispatcher_);
+  test_session_->interval_timer_ = new NiceMock<Event::MockTimer>(&dispatcher_);
+  test_session_->request_encoder_.stream_.callbacks_.clear();
+  expectClientCreate(0);
+}
+
+void GrpcHealthCheckFuzz::expectClientCreate(size_t index) {
+  TestSession& test_session = *test_session_;
+  test_session.codec_ = new NiceMock<Http::MockClientConnection>();
+  test_session.client_connection_ = new NiceMock<Network::MockClientConnection>();
+  connection_index_.push_back(index);
+  codec_index_.push_back(index);
+
+  EXPECT_CALL(dispatcher_, createClientConnection_(_, _, _, _))
```

Comment:
one thing I've been curious about, can these expects be added to `initialize`? they are used repeteadedly

Reply:
Changed to ON_CALL, and moved to initialize.